restapi/impl/db/permissions.go

```diff
@@ -215,6 +239,63 @@ func PermissionsForSubjectsAndResourceTypeMinLevel(
 	return rowsToPermissionList(rows)
 }
 
+// permissionLevelPrecedenceExpression returns a SelectBuilder representing a permission level precedence
+// expression that can be used in a where clause.
+func permissionLevelPrecedenceExpression(prefix, permissionLevel string) sq.SelectBuilder {
+	return psql.Select("precedence").
+		Prefix(fmt.Sprintf("%s (", prefix)).
+		From("permission_levels").
+		Where(sq.Eq{"name": permissionLevel}).
+		Suffix(")")
+}
+
+// AbbreviatedPermissionsForSubjectAndResouorceType lists permissions for a subject and resource type. If the
+// minLevel parameter is specified, permissions that don't meet or exceed the minimum level will be omitted
+// from the results.
+func AbbreviatedPermissionsForSubjectAndResouorceType(
+	tx *sql.Tx, subjectIDs []string, resourceTypeName string, minLevel *string,
+) ([]*models.AbbreviatedPermission, error) {
+
+	// Begin building the query.
+	builder := psql.Select(
+		"first_value(p.id) OVER w AS id",
+		"first_value(r.name) OVER w AS resource_name",
+		"first_value(rt.name) OVER w AS resource_type",
+		"first_value(pl.name) OVER w AS permission_level",
+	).Distinct().Options("ON (r.id)").
+		From("permissions p").
+		Join("permission_levels pl ON p.permission_level_id = pl.id").
+		Join("subjects s ON p.subject_id = s.id").
+		Join("resources r ON p.resource_id = r.id").
+		Join("resource_types rt ON r.resource_type_id = rt.id").
+		Where(sq.Eq{"rt.name": resourceTypeName}).
+		Where(sq.Eq{"s.subject_id": subjectIDs})
+
+	// Add the permission level expression if a minimum level was specified.
+	if minLevel != nil {
+		builder = builder.Where(permissionLevelPrecedenceExpression("pl.precedence <=", *minLevel))
+	}
+
+	// Add the window and the ORDER BY clause. The ORDER BY clause has to appear here because Squirrel doesn't have
+	// explicit support for the WINDOW clause.
```

Comment:
I suppose to be good open source citizens, we should probably open an issue in squirrel for adding this support. Not that I haven't been slacking on the same, of course.

Reply:
That's a good idea. I was also considering adding better support for subqueries in WHERE clauses.

Reply:
On a related note, the README for Squirrel says that it's functionally complete and that enhancements won't be merged. We might have to fork it or find a fork that 's being maintained.